src/actions/config.js

```diff
@@ -1,50 +1,63 @@
 import yaml from "js-yaml";
```

Comment:
Causes for change in this file:

* Config objects are now converted to immutable maps before processing.
* We must now differentiate between merging a config value into state and loading config state into the application, since state values could be merged in an arbitrary number of times before the final configuration state is ready.